src/main/java/listfiles/ListFiles.java

```diff
@@ -0,0 +1,135 @@
+                    exc.printStackTrace();
+                } catch (IOException exc) {
+                    exc.printStackTrace();
+                }
+            }
+        }
+
+        for (File file : dir.listFiles()) {
+            if (file.isDirectory()) {
+                String folder = file.getAbsolutePath();
+                String prefix = prefixString(countLevel(file.getAbsolutePath(), separator));
+                String out = folder.substring(folder.lastIndexOf(separator));
+                try {
+                    bufferedWriter.write(prefix, 0, prefix.length());
+                    bufferedWriter.write(out, 0, out.length());
+                    bufferedWriter.newLine();
+                } catch (FileNotFoundException exc) {
+                    exc.printStackTrace();
+                } catch (IOException exc) {
+                    exc.printStackTrace();
+                }
+                getFolderList(file, bufferedWriter);
+            }
+        }
+    }
+
+    static void getStatistics(File test) {
+
+        int numFiles = 0;
+        double fileNamesLength = 0.0;
+        Map<String, Integer> listOfDir = new TreeMap<>();
+
+        if (test.isFile() && test.exists()) {
+            try (FileReader input = new FileReader(test);
+                 BufferedReader bufferedReader = new BufferedReader(input);) {
+
+                String line;
+                String currentDir = "";
+
+                while ((line = bufferedReader.readLine()) != null) {
+                    if (line.indexOf(separator) >= 0) {
+                        currentDir = line.substring(line.indexOf(separator) + 1);
+                        listOfDir.put(currentDir, 0);
+                        numFiles = 0;
+                    } else {
+                        numFiles++;
+                        fileNamesLength += line.substring(line.lastIndexOf('-')+ 1).length();
+                        listOfDir.put(currentDir, numFiles);
+                    }
+                }
+            } catch (FileNotFoundException e) {
+                e.printStackTrace();
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+
+            int numberOfAllFiles = 0;
+            for (Integer number : listOfDir.values()) {
+                numberOfAllFiles += number;
+            }
+            System.out.println("Содержание файлов в директориях: ");
+            System.out.println(listOfDir);
+            System.out.println();
+
+            System.out.println("Количество директорий " + listOfDir.size());
+            System.out.println("Количество файлов во всех директориях " + numberOfAllFiles);
+            System.out.format("Среднее количество файлов в папке %.2f%n", ((double) numberOfAllFiles) / listOfDir.size());
+            System.out.format("Средняя длина файла %.2f%n", fileNamesLength / numberOfAllFiles);
+        }
+    }
+
+    public static void main(String[] args) {
+
+        inputFile = new File(args[0]);
```

Comment:
If you launch your code without arguments you'll catch
Exception in thread "main" java.lang.ArrayIndexOutOfBoundsException: 0
	at listfiles.ListFiles.main(ListFiles.java:116)
You should prevent this situation. Please give a user a second chance to type obligatory information from the console.